makefiles/features.modules.mk

```diff
@@ -0,0 +1,37 @@
+# Add modules implementing used features
+#
+# This is done after the regular dependency resolution in Makefile.dep, as
+# feature resolution depends on the used modules. As these modules however have
+# no dependencies(except for periph_common), no dependency resolution is needed.
+
+PERIPH_FEATURES := $(filter periph_%,$(FEATURES_USED))
+# all periph features correspond to a periph submodule
+# FEATURES_USED is defined in Makefile.features
+USEMODULE += $(PERIPH_FEATURES)
```

Comment:
Actually seems this can't be here a lot of platforms do use `periph_` to trigger dependency resolution:

```
git grep "ifneq (,\$(filter periph\_"
Makefile.dep:ifneq (,$(filter periph_uart_nonblocking,$(USEMODULE)))
Makefile.dep:ifneq (,$(filter periph_gpio_irq,$(USEMODULE)))
cpu/efm32/Makefile.dep:ifneq (,$(filter periph_rtc,$(USEMODULE)))
cpu/efm32/Makefile.dep:ifneq (,$(filter periph_rtt,$(USEMODULE)))
cpu/esp32/Makefile.dep:ifneq (,$(filter periph_rtc,$(USEMODULE)))
cpu/esp32/Makefile.dep:ifneq (,$(filter periph_i2c,$(USEMODULE)))
cpu/esp8266/Makefile.dep:ifneq (,$(filter periph_i2c,$(USEMODULE)))
cpu/fe310/Makefile.dep:ifneq (,$(filter periph_rtc,$(USEMODULE)))
cpu/kinetis/Makefile.dep:ifneq (,$(filter periph_rtc,$(USEMODULE)))
cpu/kinetis/Makefile.dep:ifneq (,$(filter periph_i2c,$(USEMODULE)))
cpu/native/Makefile.dep:ifneq (,$(filter periph_spi,$(USEMODULE)))
cpu/sam0_common/Makefile.dep:ifneq (,$(filter periph_uart_nonblocking,$(USEMODULE)))
cpu/stm32_common/Makefile.dep:ifneq (,$(filter periph_usbdev,$(FEATURES_USED)))
cpu/stm32_common/periph/Makefile:ifneq (,$(filter periph_i2c,$(USEMODULE)))
cpu/stm32_common/periph/Makefile:ifneq (,$(filter periph_flashpage periph_eeprom,$(USEMODULE)))
cpu/stm32_common/periph/Makefile:ifneq (,$(filter periph_wdt,$(USEMODULE)))
makefiles/features.modules.mk:ifneq (,$(filter periph_%, $(USEMODULE)))
makefiles/features.modules.mk:ifneq (,$(filter periph_init, $(USEMODULE)))
makefiles/features.modules.mk:ifneq (,$(filter periph_init, $(USEMODULE)))
pkg/fatfs/Makefile.include:ifneq (,$(filter periph_rtc,$(USEMODULE)))
sys/shell/commands/Makefile:ifneq (,$(filter periph_rtc,$(USEMODULE)))
```

In this case maybe this `makefile` does not make sense and I should just keep everything in `Makefile.dep`.

Reply:
I could also do what you did https://github.com/RIOT-OS/RIOT/blob/65c6305c5ad65d862b95edfeac5afaf7949aa043/makefiles/features_modules.inc.mk#L12-L17

But it hides what are and what are not `DEFAULT_MODULES`